structuralcodes/materials/concrete/_concreteEC2_2023.py

```diff
@@ -0,0 +1,194 @@
+"""The concrete class for EC2 2023 Concrete Material."""
+import typing as t
+
+from structuralcodes.codes import ec2_2023
+from ._concrete import Concrete
+
+
+# TODO: complete the documentation and discuss if properties/functions
+# are considered proper for the framework
+class ConcreteEC22023(Concrete):
+    """Concrete implementation for EC2 2023 Concrete."""
+
+    # Inherent concrete properties
+    _kE = t.Optional[float] = None  # noqa: N815
+    _concrete_dev_class = t.Optional[str] = None
+    _gamma_C = t.Optional[float] = None  # noqa: N815
+
+    # Computed attributes
+    _fcm = t.Optional[float] = None
+    _fctm = t.Optional[float] = None
+    _Ecm = t.Optional[float] = None
+
+    # TODO: do we agree with the optional keyword arguments here?
+    def __init__(
+        self,
+        fck: float,
+        name: t.Optional[str] = None,
+        density: float = 2400.0,
+        kE: float = 9500,
+        concrete_dev_class: str = 'CS',
```

Comment:
Could we rename `concrete_dev_class` to `strength_dev_class`? This also applies to the relevant methods in the [materials section](https://github.com/DanielGMorenaFhecor/structuralcodes/blob/238d7d562c01f6e934628bf39cae9a5a4ca43265/structuralcodes/codes/ec2_2023/_section5_materials.py).

Reply:
I also add: can it be helpful to recognize both CS, CN, CR and slow, normal, rapid?

Reply:
Perfect! I have also added the corresponding tests for the _section5_materials.py module